src/interceptors/ClientRequest/utils/normalizeClientRequestArgs.ts

```diff
@@ -110,11 +110,13 @@ export function normalizeClientRequestArgs(
   // and derive request options from it.
   else if (args[0] instanceof URL) {
     url = args[0]
-    logger.info('first argument is a URL:', url)
 
     options = resolveRequestOptions(args, url)
     logger.info('derived request options:', options)
 
+    url.pathname = options.path || url.pathname
```

Comment:
Maybe it's a bit nitpicky, but what do you think if we handle this case rather explicitly, including logging?

```ts
if (options.path) {
  logger.info('found explicit "path" option ("%s"), overriding URL pathname ("%s")', options.path, url.pathname)

  url.pathname = path
}
```

This would help in debugging any related issues a lot.

Reply:
Good idea!
I'll add it when I get back home

Reply:
Done